resource_management/azure_mgmt_compute/spec/spec_helper.rb

```diff
@@ -2,63 +2,84 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 
-require 'dotenv'
-Dotenv.load(File.join(File.dirname(__FILE__), '../../.env'))
-
+require File.join(File.dirname(__FILE__), '../../vcr_helper')
 require 'azure_mgmt_resources'
 require 'azure_mgmt_compute'
 require 'ms_rest_azure'
 require 'azure_mgmt_storage'
 require 'azure_mgmt_network'
 
-include Azure::ARM::Storage
-
 include MsRest
 include MsRestAzure
-include Azure::ARM::Resources
 include Azure::ARM::Compute
+include Azure::ARM::Compute::Models
 include Azure::ARM::Network
+include Azure::ARM::Resources
+include Azure::ARM::Storage
 
-def create_resource_group
-  resource_group_name = get_random_name('RubySDKTest_')
-  params = Azure::ARM::Resources::Models::ResourceGroup.new()
-  params.location = 'westus'
+class ResourceHelper
+  attr_accessor :compute_client
+  attr_accessor :network_client
+  attr_accessor :resource_client
+  attr_accessor :storage_client
 
-  RESOURCES_CLIENT.resource_groups.create_or_update(resource_group_name, params).value!.body
-end
+  def initialize
+    tenant_id = ENV['AZURE_TENANT_ID']
+    client_id = ENV['AZURE_CLIENT_ID']
+    secret = ENV['AZURE_CLIENT_SECRET']
+    subscription_id = ENV['AZURE_SUBSCRIPTION_ID']
 
-def delete_resource_group(name)
-  RESOURCES_CLIENT.resource_groups.delete(name).value!
-end
+    token_provider = ApplicationTokenProvider.new(tenant_id, client_id, secret)
+    credentials = TokenCredentials.new(token_provider)
 
-def get_random_name(prefix, length = 1000)
-  prefix + SecureRandom.uuid.downcase.delete('^a-zA-Z0-9')[0...length]
-end
+    @compute_client = ComputeManagementClient.new(credentials)
+    @compute_client.long_running_operation_retry_timeout = ENV['RETRY_TIMEOUT'].to_i || 30
+    @compute_client.subscription_id = subscription_id
 
-class String
-  def to_b?
-    !self.to_i.zero?
+    @network_client = NetworkManagementClient.new(credentials)
+    @network_client.long_running_operation_retry_timeout = ENV['RETRY_TIMEOUT'].to_i || 30
```

Comment:
`ENV['RETRY_TIMEOUT'].to_i || 30` will be always `0` in case of` ENV['RETRY_TIMEOUT']` is `nil`
Use `ENV.fetch('RETRY_TIMEOUT', 30).to_i`